client-c/main.cpp

```diff
@@ -98,38 +103,62 @@ int main(int argc, char **argv) {
     return 0;
   }
 
-  int sock = init(port, address);
+  // injections to KItem for initial configuration variables
+
+  static blockheader injHeaderMode               = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortMode{}, SortKItem{}}"));
+  static blockheader injHeaderSchedule           = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortSchedule{}, SortKItem{}}"));
+  static blockheader injHeaderInt                = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortInt{}, SortKItem{}}"));
+  static blockheader injHeaderBool               = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortBool{}, SortKItem{}}"));
+  static blockheader injHeaderEthereumSimulation = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortEthereumSimulation{}, SortKItem{}}"));
+
+  // create `Init` configuration variable entries
+
   static uint64_t mode = (((uint64_t)getTagForSymbolName("LblNORMAL{}")) << 32) | 1;
   inj *modeinj = (inj *)koreAlloc(sizeof(inj));
-  static blockheader hdr = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortMode{}, SortKItem{}}"));
-  modeinj->h = hdr;
+  modeinj->h = injHeaderMode;
   modeinj->data = (block*)mode;
+
   uint64_t schedule = (((uint64_t)schedule_tag) << 32) | 1;
   inj *scheduleinj = (inj *)koreAlloc(sizeof(inj));
-  static blockheader hdr2 = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortSchedule{}, SortKItem{}}"));
-  scheduleinj->h = hdr2;
+  scheduleinj->h = injHeaderSchedule;
   scheduleinj->data = (block*)schedule;
+
+  int sock = init(port, address);
   zinj *sockinj = (zinj *)koreAlloc(sizeof(zinj));
-  static blockheader hdr3 = getBlockHeaderForSymbol(getTagForSymbolName("inj{SortInt{}, SortKItem{}}"));
-  sockinj->h = hdr3;
+  sockinj->h = injHeaderInt;
   mpz_t sock_z;
   mpz_init_set_si(sock_z, sock);
   sockinj->data = move_int(sock_z);
+
+  uint32_t shutdownable_tag = shutdownable ? getTagForSymbolName("\\dv{SortBool{}}(\"true\")")
+                                           : getTagForSymbolName("\\dv{SortBool{}}(\"false\")");
+  static uint64_t shutdownableBody = (((uint64_t)shutdownable_tag) << 32) | 1;
+  boolinj *shutdownableinj = (boolinj *)koreAlloc(sizeof(boolinj));
+  shutdownableinj->h = injHeaderBool;
+  shutdownableinj->data = (block *)shutdownableBody;
```

Comment:
@dwightguth my guess is that `shutdownable_tag` is not being defined correctly, can you help?